content/docs/rfcs/38/README.md

```diff
@@ -0,0 +1,556 @@
+    ;; total number of YES (i.e. positive) votes for the truth of the proposal
+    total_positive 
+      <-- 0
+    ;; current number of nodes to attempt to query in a round
+    k 
+      <-- k_original
+    ;; the current query round     
+    round
+      <-- 0
+
+
+###  Query 
+
+A node selects $k$ nodes randomly from the current locally known
+complete set of peers in the network.  
+
+A query is sent to each neighbor with the node's current `opinion` of
+the proposal.
+
+Each node replies with their current opinion on the proposal.
+
+See [Interoperability section](#interoperability) for details on the
+(evolving) semantics and syntax of the "on the wire" representation of
+this query.
+
+When the query finishes, the node now initializes the following two
+values:
+
+    new_votes 
+      <-- |total vote replies received in this round to the current query|
+    positive_votes 
+      <-- |YES votes received from the query| 
+    
+### Computation    
+
+The node runs the `new_votes` and `positive_votes` parameters received
+in the query round through the following algorithm:
+
+    total_votes 
+      += new_votes
+    total_positive 
+      += positive_votes
+    confidence 
+      <-- total_votes / (total_votes + look_ahead) 
+    total_evidence 
+      <-- total_positive / total_votes
+    new_evidence 
+      <-- positive_votes / new_votes
+    evidence 
+      <-- new_evidence * ( 1 - confidence ) + total_evidence * confidence 
+    alpha 
+      <-- doubt * ( 1 - confidence ) + certainty * confidence 
+    
+### Opinion 
+
+The node updates its local opinion on the consensus proposal by
+examining the relationship between the evidence accumulated for a
+proposal with the confidence encoded in the `alpha` parameter:
+
+    IF
+      evidence > alpha
+    THEN ;; The node adopts the opinion YES on the proposal
+      opinion <-- YES
+    ELSE IF       
+      evidence < 1 - alpha
+    THEN;; The node adopts the opinion NO on the proposal 
+      opinion <-- NO
+       
+If the opinion of the node is `UNDECIDED` after evaluating the opinion
+phase, the number of uniform randomly queries nodes is adjusted by
+multiplying the neighbors `k` by the `k_multiplier` up to the limit
+of `k_max_multiplier` query size increases.
+    
+    ;; possibly increase number nodes to uniformly randomly query in next round
+    WHEN
+       opinion == UNDECIDED
+    AND 
+       k < k_original * k_multiplier ^ max_k_multiplier
+    THEN 
+       k <-- k * k_multiplier
+
+###  Decision 
+
+After the OPINION phase is executed, the current value of `confidence`
+is considered: if `confidence` exceeds a threshold derived from the
+network size and directly related to the total votes received, an
+honest node marks the decision as final, and always returns this
+opinion is response to further queries from other nodes on the
+network.
+ 
+    IF 
+      confidence > confidence_threshold
+    OR 
+      round > max_rounds
+    THEN
+      finalized <-- T
```

Comment:
This finalized flag could be the `desion` itself.

Reply:
Hmm, I haven't specified the Termination Conditions.  I was in the middle of refactoring it out in a section to describe modeling/running in async/sync modes.  I'll TODO RSN.